### PR TITLE
fix: Deploy services after database migrations.

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: cloud/aws/deploy/sql.go [hunk 1]
```diff
@@ -225,8 +225,25 @@ func (a *NitricAwsPulumiProvider) rds(ctx *pulumi.Context) error {
...
+
+	Migrated pulumi.BoolOutput
+}
+
 // Sqldatabase - Implements PostgresSql database deployments use AWS Aurora
 func (a *NitricAwsPulumiProvider) SqlDatabase(ctx *pulumi.Context, parent pulumi.Resource, name string, config *deploymentspb.SqlDatabase) error {
+	a.SqlDatabases[name] = &RdsDatabase{
+		Name: name,
+	}
+	err := ctx.RegisterComponentResource("nitric:cloud:aws:SqlDatabase", name, a.SqlDatabases[name], pulumi.Parent(parent))
```
Comment (HomelessDinosaur) [inline cloud/aws/deploy/sql.go:242]:
```suggestion
	err := ctx.RegisterComponentResource("nitricaws:SqlDatabase", name, a.SqlDatabases[name], pulumi.Parent(parent))
```

File: cloud/aws/deploy/service.go [hunk 2]
```diff
@@ -209,6 +210,22 @@ func (a *NitricAwsPulumiProvider) Service(ctx *pulumi.Context, parent pulumi.Res
 		// Include the base database cluster URI for the runtime to resolve databases based on their name
 		envVars["NITRIC_DATABASE_BASE_URL"] = pulumi.Sprintf("postgres://%s:%s@%s:%s", "nitric", a.DbMasterPassword.Result,
 			a.DatabaseCluster.Endpoint, "5432")
+
+		sqlDatasesMigrated := lo.Map(lo.Values(a.SqlDatabases), func(item *RdsDatabase, idx int) interface{} {
```
Comment (davemooreuws) [inline cloud/aws/deploy/service.go:214]:
```suggestion
		sqlDatabasesMigrated := lo.Map(lo.Values(a.SqlDatabases), func(item *RdsDatabase, idx int) interface{} {
```

File: cloud/aws/deploy/service.go [hunk 3]
```diff
@@ -209,6 +210,22 @@ func (a *NitricAwsPulumiProvider) Service(ctx *pulumi.Context, parent pulumi.Res
...
+			return item.Migrated
+		})
+
+		databasesMigrated := pulumi.All(sqlDatasesMigrated...).ApplyT(func(migrated []interface{}) bool {
```
Comment (davemooreuws) [inline cloud/aws/deploy/service.go:218]:
```suggestion
		databasesMigrated := pulumi.All(sqlDatabasesMigrated...).ApplyT(func(migrated []interface{}) bool {
```

**Review by davemooreuws — CHANGES_REQUESTED** [review]
Looks good, just a typo I think

**Review by davemooreuws — APPROVED** [review]

**Review by HomelessDinosaur — APPROVED** [review]

**Pull request merged**